src/crystal/main.cr

```diff
@@ -53,6 +58,12 @@ module Crystal
     ex.inspect_with_backtrace STDERR if ex
     STDOUT.flush
     STDERR.flush
+
+    # Restore blocking state of STDIN, STDOUT and STDERR
```

Comment:
I'd advise adding `FIXME` flag to indicate that it'll fail in case of segfault.

Reply:
Would it be worth it to also try to restore blocking behavior in case of SIGSEGV, SIGTERM, and SIGINT?